liboptv/tests/check_parameters.c

```diff
@@ -87,11 +118,15 @@ START_TEST(test_read_control_par)
     cpar_correct.mm->n1 = 1;
     cpar_correct.mm->n2[0] = 1.49;
     cpar_correct.mm->n3 = 1.33;
-    cpar_correct.mm->d[0] = 5;
+    cpar_correct.mm->d[0] = 52;
+    cpar_correct.mm->n2[1] = 0;
+    cpar_correct.mm->n2[2] = 0;
+    cpar_correct.mm->d[1] = 0;
+    cpar_correct.mm->d[2] = 0;
     
   
     cpar = read_control_par("testing_fodder/parameters/ptv.par");
-    fail_unless(compare_control_par(cpar, &cpar_correct));
+    fail_unless(compare_control_par(cpar, &cpar_correct) == 1);
```

Comment:
I just checked and the documentation only promises a True return value, not necessarily one. So the old line was more correct.